File: src/plugin/inter/FriendChatInterfacePlugin.java
package plugin.inter;

import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.content.plugin.type.InterfacePlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 8/31/2017
 */
public class FriendChatInterfacePlugin extends InterfacePlugin {
	
	@Override
	public boolean handle(Player player, int interfaceId, int componentId, int itemId, int slotId, int packetId) {
		player.getContactManager().handleFriendChatButtons(interfaceId, componentId, packetId);
		return true;
	}
	
	@Override
	public void register() {
		registerInterfacePlugin(1108, 1109, 1110);
	}
}

File: src/main/kotlin/plugin/rsinterface/FriendChatInterfacePlugin.kt
package plugin.rsinterface

import org.redrune.game.content.plugin.type.InterfacePlugin
import org.redrune.game.entity.actor.player.Player

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 8/31/2017
 */
class FriendChatInterfacePlugin : InterfacePlugin {
    override fun handle(
        player: Player,
        interfaceId: Int,
        componentId: Int,
        itemId: Int,
        slotId: Int,
        packetId: Int,
    ): Boolean {
        player.contactManager.handleFriendChatButtons(interfaceId, componentId, packetId)
        return true
    }

    override fun register() {
        registerInterfacePlugin(1108, 1109, 1110)
    }
}